### PR TITLE
Port to python3 and beautifulsoup4

### PR DESCRIPTION
**Pull request by Phenitei**
Because I'm trying to keep python2 away from my computer!

Cheers!

### CODE REVIEW

File: bin/menu [hunk 1]
```diff
@@ -16,23 +16,23 @@ import epfl.menu
...
 
-sys.stdout = codecs.getwriter(locale.getpreferredencoding())(sys.stdout) 
+sys.stdout = codecs.getwriter(locale.getpreferredencoding())(sys.stdout.detach())
 
 def print_parsed_menu(resto_regexp='', supper=False):
     restos = epfl.menu.get(supper)
-    clean_resto_list = filter(lambda a: re.search(resto_regexp, a[0], flags=re.IGNORECASE), restos.iteritems())
-    for (resto_name, resto) in  clean_resto_list:
+    clean_resto_list = [a for a in iter(restos.items()) if re.search(resto_regexp, a[0], flags=re.IGNORECASE)]
+    for (resto_name, resto) in clean_resto_list:
```
Comment (gcmalloc) [inline bin/menu]: no need for parenthesis around the unpack.

Comment (Phenitei) [inline bin/menu]: True enough, feel free to remove them if you find it too heavy!

**Comment by Phenitei** [comment]
More importantly, I just realized I forgot to change the shebang at the beginning of the bin/menu file - that's something to fix as well.

**Comment by gcmalloc** [comment]
have a look at https://github.com/gcmalloc/epfl-menu/pull/3, and tell me if that's you wanted to achieve.

**Comment by Phenitei** [comment]
Perfect, thanks! Closing this then.

**Comment by Phenitei** [comment]
Perfect, thanks! Closing this then.

**Pull request closed without merging**